services/ui-src/src/routes/ReviewSubmit/ReviewSubmit.tsx

```diff
@@ -1,29 +1,215 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Image,
+  Heading,
+  Text,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { Icon, Modal, ReportContext, ReportPage, Sidebar } from "components";
+// types
+import { ReportStatus, UserRoles } from "types";
+// utils
+import { useUser } from "utils";
+// form data
+import reviewVerbiage from "verbiage/pages/mcpar/mcpar-review-and-submit";
+import successVerbiage from "verbiage/pages/mcpar/mcpar-successful-submit";
+// assets
+import checkIcon from "assets/icons/icon_check_circle.png";
 
 export const ReviewSubmit = () => {
-  const { intro } = verbiage;
+  const { reportStatus, fetchReportStatus, updateReportStatus } =
+    useContext(ReportContext);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  // get user's state
+  const { user } = useUser();
+  const { state, userRole } = user ?? {};
+  const reportYear = "2022";
+  const reportKey = `${state}${reportYear}`;
+
+  // TODO: get real program name per report
+  const programName = "tempName";
+
+  const reportDetails = {
+    key: reportKey,
+    programName: programName,
+  };
+
+  useEffect(() => {
+    fetchReportStatus(reportDetails);
+  }, []);
+
+  const submitForm = () => {
+    // TODO: Add check to make sure user filled out the form
+    if (userRole === UserRoles.STATE_USER || userRole === UserRoles.STATE_REP) {
+      updateReportStatus(reportDetails, ReportStatus.COMPLETED);
+    }
+    // TODO: Handle Error for other users coming on to the page
+    onClose();
+  };
+
   return (
     <ReportPage data-testid="review-and-submit-view">
+      <Flex sx={sx.pageContainer}>
+        <Sidebar />
+        {reportStatus?.status == ReportStatus.COMPLETED ? (
+          <SuccessMessage
+            programName={programName}
+            date={reportStatus?.lastAltered}
+            givenName={user?.given_name}
+            familyName={user?.family_name}
+          />
+        ) : (
+          <ReadyToSubmit
+            submitForm={submitForm}
+            isOpen={isOpen}
+            onOpen={onOpen}
+            onClose={onClose}
+          />
+        )}
+      </Flex>
+    </ReportPage>
+  );
+};
+
+const ReadyToSubmit = ({
+  submitForm,
+  isOpen,
+  onOpen,
+  onClose,
+}: ReadyToSubmitProps) => {
+  const { intro, modal, pageLink } = reviewVerbiage;
+
+  return (
+    <Flex sx={sx.contentContainer} data-testid="ready-view">
       <Box sx={sx.leadTextBox}>
         <Heading as="h1" sx={sx.headerText}>
           {intro.header}
         </Heading>
-        <Text>{intro.body}</Text>
+        <Box sx={sx.infoTextBox}>
+          <Text sx={sx.infoHeading}>{intro.infoHeader}</Text>
+          <Text>{intro.info}</Text>
+        </Box>
       </Box>
-    </ReportPage>
+      <Flex sx={sx.submitContainer}>
+        <Button
+          type="submit"
+          colorScheme="colorSchemes.primary"
+          rightIcon={<Icon icon="arrowRight" />}
+          onClick={() => onOpen()}
```

Comment:
Because you aren't passing any props to `onOpen` you can simplify this. 
```suggestion
          onClick={onOpen}
```

Reply:
I thought so too but that pops up with an error:
 `Type 'Function' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.`

Reply:
Best way to solve this would be something like this then:
`onClick={onOpen as MouseEventHandler}`

Reply:
So easy I should have seen that 🙃